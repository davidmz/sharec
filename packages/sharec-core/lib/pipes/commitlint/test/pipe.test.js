const { fixtures } = require('testUtils')
const { pipe } = require('../pipe')

describe('strategies > pipes > commitlint > pipe', () => {
  describe('JSON', () => {
    const commitlintFxt = fixtures('commitlint/json/00-base')

    it('should merge configs', () => {
      expect(pipe('.commitlintrc.json')(commitlintFxt)).toMatchSnapshot()
    })
  })

  describe('YAML', () => {
    const commitlintFxt = fixtures('commitlint/yaml/00-base')

    it('should merge configs', () => {
      expect(pipe('.commitlintrc.yaml')(commitlintFxt)).toMatchSnapshot()
    })
  })
})
