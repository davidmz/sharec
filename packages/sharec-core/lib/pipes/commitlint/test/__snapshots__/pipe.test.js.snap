// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strategies > pipes > commitlint > pipe JSON should merge configs 1`] = `
"{
  \\"parserPreset\\": \\"other-parser\\",
  \\"rules\\": {
    \\"body-leading-blank\\": [
      2,
      \\"never\\"
    ],
    \\"body-max-line-length\\": [
      2,
      \\"always\\",
      100
    ],
    \\"footer-leading-blank\\": [
      1,
      \\"always\\"
    ],
    \\"type-enum\\": [
      2,
      \\"always\\",
      [
        \\"feat\\",
        \\"fix\\",
        \\"perf\\",
        \\"refactor\\",
        \\"revert\\",
        \\"style\\",
        \\"test\\"
      ]
    ],
    \\"footer-max-line-length\\": [
      2,
      \\"always\\",
      100
    ],
    \\"header-max-length\\": [
      2,
      \\"always\\",
      100
    ]
  }
}"
`;

exports[`strategies > pipes > commitlint > pipe YAML should merge configs 1`] = `
"parserPreset: other-parser
rules:
  body-leading-blank:
    - 2
    - never
  body-max-line-length:
    - 2
    - always
    - 100
  footer-leading-blank:
    - 1
    - always
  type-enum:
    - 2
    - always
    - - feat
      - fix
      - perf
      - refactor
      - revert
      - style
      - test
  footer-max-line-length:
    - 2
    - always
    - 100
  header-max-length:
    - 2
    - always
    - 100"
`;
